source/Cosmos.System2_Plugs/System/IO/FileInfoImpl.cs

```diff
@@ -0,0 +1,21 @@
+﻿//#define COSMOSDEBUG
+using System.IO;
+using IL2CPU.API.Attribs;
+using Cosmos.System;
+using Cosmos.System.FileSystem.VFS;
+
+namespace Cosmos.System2_Plugs.System.IO
+{
+    [Plug(Target = typeof(FileInfo))]
+    public static class FileInfoImpl
+    {
+        /* Optimize this: CosmosVFS should expose an attribute without the need to open the file for reading... */
```

Comment:
I think that the FAT driver has support for entry sizes. The VFS has to provide a way of getting the size, but it doesn't need to be done in this PR.

Reply:
Yes I think so, this can be a "starting point".